src/atom.rs

```diff
@@ -211,70 +235,94 @@ impl Atom {
     }
 }
 
-impl Default for Atom {
+impl<Static: StaticAtomSet> Default for Atom<Static> {
+    #[inline]
     fn default() -> Self {
-        atom!("")
+        Atom {
+            unsafe_data: pack_static(Static::empty_string_index()),
+            phantom: PhantomData
+        }
+    }
+}
+
+impl<Static: StaticAtomSet> Hash for Atom<Static> {
+    #[inline]
+    fn hash<H>(&self, state: &mut H) where H: Hasher {
+        self.unsafe_data.hash(state)
+    }
+}
+
+impl<Static: StaticAtomSet> Eq for Atom<Static> {}
+
+// NOTE: This impl requires that a given string must always be interned the same way.
+impl<Static: StaticAtomSet> PartialEq for Atom<Static> {
+    #[inline]
+    fn eq(&self, other: &Self) -> bool {
+        self.unsafe_data == other.unsafe_data
     }
 }
 
-impl PartialEq<str> for Atom {
+impl<Static: StaticAtomSet> PartialEq<str> for Atom<Static> {
     fn eq(&self, other: &str) -> bool {
         &self[..] == other
     }
 }
 
-impl PartialEq<Atom> for str {
-    fn eq(&self, other: &Atom) -> bool {
+impl<Static: StaticAtomSet> PartialEq<Atom<Static>> for str {
+    fn eq(&self, other: &Atom<Static>) -> bool {
         self == &other[..]
     }
 }
 
-impl PartialEq<String> for Atom {
+impl<Static: StaticAtomSet> PartialEq<String> for Atom<Static> {
     fn eq(&self, other: &String) -> bool {
         &self[..] == &other[..]
     }
 }
 
-impl<'a> From<Cow<'a, str>> for Atom {
+impl<'a, Static: StaticAtomSet> From<Cow<'a, str>> for Atom<Static> {
     #[inline]
-    fn from(string_to_add: Cow<'a, str>) -> Atom {
-        let unpacked = match STATIC_ATOM_SET.get_index_or_hash(&*string_to_add) {
-            Ok(id) => Static(id as u32),
-            Err(hash) => {
-                let len = string_to_add.len();
-                if len <= MAX_INLINE_LEN {
-                    let mut buf: [u8; 7] = [0; 7];
-                    copy_memory(string_to_add.as_bytes(), &mut buf);
-                    Inline(len as u8, buf)
-                } else {
-                    Dynamic(STRING_CACHE.lock().unwrap().add(string_to_add, hash) as *mut ())
-                }
+    fn from(string_to_add: Cow<'a, str>) -> Self {
+        let static_set = Static::get();
+        let hash = phf_shared::hash(&*string_to_add, static_set.key);
```

Comment:
It would have to be a method of a public trait, and thus changing it (if we need to in the future) would be a breaking change.